Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp

```diff
@@ -442,6 +484,44 @@ namespace AZ
                     // wait until swapchain has been created before setting fullscreen state
                     AzFramework::WindowSize resolution;
                     float scale = AZStd::max(static_cast<float>(r_renderScale), 0.f);
+
+#if defined(CARBONATED)
+                    if (r_resolutionMode > 0u)
+                    {
+                        resolution.m_width = static_cast<uint32_t>(r_width);
+                        resolution.m_height = static_cast<uint32_t>(r_height);
+                    }
+                    else
+                    {
+                        const auto& windowSize = m_nativeWindow->GetClientAreaSize();
+                        resolution.m_width = static_cast<uint32_t>(windowSize.m_width);
+                        resolution.m_height = static_cast<uint32_t>(windowSize.m_height);
+                    }
```

Comment:
The if-else above looks the same as in the #else (not defined CARBONATED) section. It makes sense to to leave this code piece as is, then add just #if - #endif CARBONATED section with no #else.

Reply:
No, take a closer look - the #else section  applies the scale in the if/else, and here I just assign the width & height pre-scale, applying the scale later.

Reply:
Indeed, missed that.